creator-node/src/snapbackSM/snapbackSM.js

```diff
@@ -974,72 +979,88 @@ class SnapbackSM {
+          secondaryCaughtUpToPrimary = true
           break
         }
       } catch (e) {
-        this.log(`${logMsgString} || Error: ${e.message}`)
+        this.logError(`${logMsgString} || Error: ${e.message}`)
       }
 
       // Delay between retries
       await utils.timeout(SyncMonitoringRetryDelayMs, false)
     }
 
+    const monitoringTimeMs = Date.now() - startTimeMs
+
     /**
-     * As Primary for user, record syncRequest outcomes to all secondaries
-     * For now, ignore syncRequests where a secondary is behind by more than `MaxExportClockValueRange` as
-     *    primary doesn't currently have sufficient tracking
+     * As Primary for user, record SyncRequest outcomes to all secondaries
+     * Also check whether additional sync is required
      */
-    if (!additionalSyncRequired && !maxExportRangeExceeded) {
+    let additionalSyncIsRequired
+    if (secondaryCaughtUpToPrimary) {
+      await SecondarySyncHealthTracker.recordSuccess(secondaryUrl, userWallet, syncType)
+      additionalSyncIsRequired = false
+      this.logDebug(`${logMsgString} || Sync completed in ${monitoringTimeMs}ms`)
+
+      // Secondary completed sync but is still behind primary since it was behind by more than max export range
+      // Since syncs are all-or-nothing, if secondary clock has increased at all, we know it successfully completed sync
+    } else if (finalSecondaryClock > initialSecondaryClock) {
       await SecondarySyncHealthTracker.recordSuccess(secondaryUrl, userWallet, syncType)
+      additionalSyncIsRequired = true
+      this.log(`${logMsgString} || Secondary successfully synced from clock ${initialSecondaryClock} to ${finalSecondaryClock} but hasn't caught up to Primary. Enqueuing additional syncRequest.`)
+
+      // (1) secondary did not catch up to primary AND (2) secondary did not complete sync
     } else {
       await SecondarySyncHealthTracker.recordFailure(secondaryUrl, userWallet, syncType)
+
+      const secondaryUserSyncFailureCountToday = await SecondarySyncHealthTracker.getSecondaryUserSyncFailureCountToday(
+        secondaryUrl, userWallet, syncType
+      )
+
+      const SecondaryUserSyncDailyFailureCountThreshold = this.nodeConfig.get('secondaryUserSyncDailyFailureCountThreshold')
+
+      if (secondaryUserSyncFailureCountToday > SecondaryUserSyncDailyFailureCountThreshold) {
+        additionalSyncIsRequired = false
+        this.logError(`${logMsgString} || Secondary failed to progress from clock ${initialSecondaryClock} and has met SecondaryUserSyncDailyFailureCountThreshold (${SecondaryUserSyncDailyFailureCountThreshold}). Will not enqueue further syncRequests today.`)
+      } else {
+        additionalSyncIsRequired = true
+        this.logError(`${logMsgString} || Secondary failed to progress from clock ${initialSecondaryClock}. Enqueuing additional syncRequest.`)
+      }
     }
 
-    return additionalSyncRequired
+    return additionalSyncIsRequired
```

Comment:
good thinking with this flow here! it makes sense what we're doing here to determine if an additional sync is required, and whether or not to mark a sync as a success or fail. I think we can simplify this even more like this with less variable declarations + more readability. what do you think?

```
let initialSecondaryClockValue, latestSecondaryClockValue, additionalSyncIsRequired
while (Date.now() < maxMonitoringTimeMs) {
	try {
		latestSecondaryClockValue = await CreatorNode.getClockValue(secondaryUrl, userWallet)
		if (initialSecondaryClockValue === undefined) initialSecondaryClockValue = latestSecondaryClockValue
		
		if (latestSecondaryClockValue >= primaryClockValue) {
			await SecondarySyncHealthTracker.recordSuccess(secondaryUrl, userWallet, syncType)
      		additionalSyncIsRequired = false
			break
		}
	} catch (e) {}

	await utils.timeout(SyncMonitoringRetryDelayMs, false)
}	

if (additionalSyncIsRequired) return additionalSyncIsRequired

// If reached this, implies that the secondary did not progress enough or did not progress at all during sync

const secondaryUserSyncFailureCountToday = await SecondarySyncHealthTracker.getSecondaryUserSyncFailureCountToday(
	secondaryUrl, userWallet, syncType
)

// If max attempts were reached, do not continue enqueuing syncs 
if (secondaryUserSyncFailureCountToday > SecondaryUserSyncDailyFailureCountThreshold) {
	await SecondarySyncHealthTracker.recordFailure(secondaryUrl, userWallet, syncType)
	additionalSyncIsRequired = false
} else {
	if (initialSecondaryClockValue === latestSecondaryClockValue) {
		// No progress was made
		await SecondarySyncHealthTracker.recordFailure(secondaryUrl, userWallet, syncType)
	} else {
		// Some progress was made
		await SecondarySyncHealthTracker.recordSuccess(secondaryUrl, userWallet, syncType)
	}
	additionalSyncIsRequired = true
}

return additionalSyncIsRequired
```

Reply:
i tried writing this out and i don't think its actually better than current version - i like keeping all outcome recording in one place, and while the current logic might be **slightly** more verbose i think it is cleaner

i can see how above version is more concise and might be more readable to you but i think its very minor

gonna leave it as is if cool with you

Reply:
yup, sg 👍